### ISSUE TITLE
UI::Notification utterly useless

### ISSUE THREAD

**Issue by Eneroth3** [SketchUp/api-issue-tracker#482]
Already when the notifications were added I brought up how the delay until it disappears is a fraction of what it takes to notice the notification and start reading it. Making an informed decision based on the text is impossible.

I propose either the delay to be increased tenfold, or this accessibility disaster to be culled from SketchUp and removed from the API.

**Comment by Fredosixx** [comment]
I fully agree. The delay is quite short. The only way to keep reading is to move the cursor within the notification window. Then, it stays open.

A better design would have been
 1) to control the delay as a parameter
 2) or to provide the capability to display notification permanently and a hide() method to close it from Ruby.

While we are at it

1) **No control over the size of the text**. On my laptop, it displays quite small.

![image](https://user-images.githubusercontent.com/57754157/82901412-2aadd900-9f5e-11ea-8dbd-26997d8d8884.png)

2) **It would have been better to specify the buttons in parameters** (so we can have more than 2) and have a unique callback for which button was pressed.

3) **The parameter _extension_ is totally useless**. First the extension name does not appear anywhere in the notification (neither title bar or body). Second, this is typically not an information that all portions of an extension code has at hand easily.

I think I will return to my practice of HTML dialog, with a timer to close. More lines of code, but at least we can control the content.

**Comment by Fredosixx** [comment]
And ThomThom, please relay our frustation not to be associated, or even just consulted, for new features in the API or related to development. This would avoid a minimum of disappointment, as it has been for:
  - HTML dialog, where all questions of focus, sizing and even migration from WebDialog were not covered
  - ImageRep, where there are a few missing methods, but critical for performance
  - the Observers, which are changed but never fixed or redesigned
  - ...and of course, the whole of EWH, which completely missed the point

This is really a pity not to ask...

**Comment by Eneroth3** [comment]
1. I think it's fine to have a consistent delay not controlled by each developer, but it should be controlled by the user in Preferences > Accessibility. And the default value needs to be something reasonable (this is an easy fix that doesn't require any UI changes and can be a first step).

2. I'd personally love to have an infinite delay and actively close each notification, but again I think that's a user choice. Some people read quickly, even under pressure.  If just one extension's notifications stay infinitely and others disappear that could easily feel like a bug to the user.

1?. For consistency I don't think this should be controlled for individual notifications, but the size could increase for all notifications.

2?. Something like a general `add_button("Text") { callback... }`  method could also work.

3. I get why this was included as there has been a problem in the past with WebDialogs popping up without saying what extension they belonged to, and how the API from the start needs to lay the foundation for potential future functionality to allow it to be added without breaking existing extensions, but currently this parameter seems to be unused. Just adding the extension name as a title in the titlebar thingy of the notification would make it a lot easier to quickly interpret the message.

**Comment by Fredosixx** [comment]
I don't think it costs more to have a configurable delay by notification.

And this can be exploited for different situations. For instance, as a knid of minihelp in some tools (instead of using tooltips or status bar, always too short).

And why not add the capability to pass HTML text for better rendering...

The main interest of notification over html dialogs is that they don't steal the focus.

**Comment by Eneroth3** [comment]
I don't think Notifications should be used for other things than notifications. Such use is unexpected for the user and confusing. For help info the Instructor already exists and for having HtmlDialogs that don't steel focus, there could be a parameter at the HtmlDialog to not have it steal focus.

**Comment by DanRathbun** [comment]
I'd prefer the notifications to be displayed in a stack within an inspector window. After a user settable delay, the window automatically collapses. This allows the user to ignore the whole stack or return to the stack (by expanding the window again,) and dealing with the notifications at their leisure.
